jenkins-formula/jenkins/agent.sls

```diff
@@ -0,0 +1,47 @@
+{#-
+Salt state file for managing Jenkins Agents
+Copyright (C) 2023 SUSE LLC <georg.pfuetzenreuter@suse.com>
+
+This program is free software: you can redistribute it and/or modify
+it under the terms of the GNU General Public License as published by
+the Free Software Foundation, either version 3 of the License, or
+(at your option) any later version.
+
+This program is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+GNU General Public License for more details.
+
+You should have received a copy of the GNU General Public License
+along with this program.  If not, see <https://www.gnu.org/licenses/>.
+-#}
+
+{%- from 'jenkins/map.jinja' import agent -%}
+
+jenkins_agent_packages:
+  pkg.installed:
+    - name: jenkins-agent
+
+{%- if 'sysconfig' in agent %}
+jenkins_agent_sysconfig:
+  file.keyvalue:
+    - name: /etc/sysconfig/jenkins-agent
+    - key_values:
+      {%- for k, v in agent.sysconfig.items() %}
+        {{ k | upper }}: '"{{ v }}"'
+      {%- endfor %}
+    - ignore_if_missing: {{ opts['test'] }}
```

Comment:
So if agent.sls is not running in test mode, and for some reason /etc/sysconfig/jenkins-agent doesn't exist, file.keyvalue will fail. This is not necessarily a bad thing (because the package is expected to create the sysconfig file), I Just want to make sure this doesn't go unnoticed.

Also, would adding `append_if_not_found` make sense?

Reply:
Yes, `ignore_if_missing: false` should be equal to it not being present. In another formula I wrapped it in an if-clause, when I realized the one-liner should do as well.

I think `append_if_not_found` is not needed in this particular configuration as I trust my package to provide the expected keys (and for it to ignore any others one might add).

Reply:
As discussed in IRC, added `append_if_not_found`. :-)